### DISCUSSION TITLE
Do some check after each step before to run the next step

### DISCUSSION THREAD

**Discussion by incoming-th** [codeceptjs/CodeceptJS#4242] [Q&A]
Hi,

To limit the failure of my tests, I want to wait for something after each step. Only when this check is done, script can continue with next step, etc.

For example, after each step I would like to wait for any class ".loading" to not be present in the DOM before to continue to next step.

I tried to use the event hook but it seems to not wait for the events to finish and it does not block the next step:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/fbf1f26f-91f3-4f51-9397-938e92ea8c33)

Not sure if I am doing this properly, but waiting for this class to not be present will grandly increase the success of the tests. I tried with all steps events already but no luck so far. So my guess is that possible but somewhere else.

Thanks

**Comment by jancorvus** [comment]
You should wrap your code in recorder
```ts
  event.dispatcher.on(event.step.finished, async (step) => {
    await recorder.add('wait to hide after each step...', async () => {
      await step.helper.waitToHide('.loading');
    }, true);
  });
```

**Reply by incoming-th** [reply]
That's interesting, I never really understood the differences and relation between helper and recorder.

Is the helper giving instruction to the recorder, or is the recorder just recording action from the helper? I don't see the full picture I believe.

**Comment by incoming-th** [comment]
I believe this is not possible via the events Hook, but I was able to make this happen by using the _afterStep event in a custom Helper:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/6036166b-dbec-481b-94ed-2b59105d0005)
